draft-ietf-mls-protocol.md

```diff
@@ -1287,10 +1294,16 @@ state as follows:
   the null optional value
 * Update the ratchet tree by replacing nodes in the direct
   path from the removed leaf using the information in the Remove message
-* Reduce the size of the roster and the tree until the rightmost
-  element roster element and leaf node are non-null
 * Update the ratchet tree by setting to blank all nodes in the
-  direct path of the removed leaf
+  direct path of the removed leaf, and also setting the root node
+  to blank
+* Truncate the roster such that the last roster element is
+  non-null (there must be at least one such an element, since a
+  pre-Add GroupState cannot process a Remove, a non-pre-Add GroupState
+  must have the current member in the roster, and self-removal is
+  prohibited)
+* Truncate the tree such that the rightmost non-blank leaf is the
+  last node of the tree (clearing the tree if no such leaf exists)
```

Comment:
We're coming down to kind of a matter of taste here, since I don't really like discussing things that are impossible :)  How about we compromise with the following:

* Leave these clarifications out of the instruction bullets
* Add a para afterward of the form "Note that there is always at least one non-blank..."

Reply:
agreed. That will also give room to mention that the same argument holds for tree truncation